apps/framework-cli/src/framework/core/infrastructure_map.rs

```diff
@@ -0,0 +1,213 @@
+use hyper::http::Method;
+use std::{collections::HashMap, time::Duration};
+
+use crate::framework::data_model::{
+    config::EndpointIngestionFormat,
+    schema::{DataModel, Table},
+};
+
+use super::primitive_map::PrimitiveMap;
+
+type TopicId = String;
+type APIId = String;
+
+#[derive(Debug)]
+pub struct Topic {
+    pub version: String,
+    pub name: TopicId,
+
+    pub retention_period: Duration,
+    pub source_primitive: PrimitiveSignature,
+}
+
+impl Topic {
+    pub fn from_data_model(data_model: &DataModel) -> Self {
+        Topic {
+            name: data_model.name.clone(),
+            version: data_model.version.clone(),
+            // TODO configure this from data model
```

Comment:
if we write `configure this from DataModelConfig` it might be more searchable

Reply:
Good call